packages/flutter/lib/src/material/badge.dart

```diff
@@ -275,7 +323,104 @@ class _RenderBadge extends RenderAligningShiftedBox {
     final double badgeSize = child!.size.height;
     final Alignment resolvedAlignment = alignment.resolve(textDirection);
     final BoxParentData childParentData = child!.parentData! as BoxParentData;
-    childParentData.offset = offset + resolvedAlignment.alongOffset(Offset(size.width - badgeSize, size.height - badgeSize));
+    Offset badgeLocation = offset + resolvedAlignment.alongOffset(Offset(size.width - widthOffset, size.height));
```

Comment:
Just a question here, the logic here and the change on `defaultOffset` seem cause some breaking changes on vertical location. Do you think if it would make sense to keep the logic here so that only the size of the badge is fixed and location won't change?